odoo/addons/base/models/ir_model.py

```diff
@@ -721,6 +721,8 @@ def _drop_column(self):
                     rel_name = field.relation_table or (is_model and model._fields[field.name].relation)
                     tables_to_drop.add(rel_name)
             if field.state == 'manual' and is_model:
+                if field.name == model._rec_name:
+                    model._pop_field('display_name')
```

Comment:
From this moment on, `model._rec_name` is invalid, and that's what `display_name` will use for its dependencies.
What about fixing `model._rec_name` instead?
```suggestion
                if field.name == model._rec_name:
                    type(model)._rec_name = 'id'
```